dependencies/src/test/java/com/google/cloud/tools/opensource/dependencies/RepositoryUtilityTest.java

```diff
@@ -57,34 +69,73 @@ public void testReadBom_coordinates() throws ArtifactDescriptorException {
     Assert.assertEquals("com.google.cloud:google-cloud-bom:0.61.0-alpha", bom.getCoordinates());
   }
 
+  private static final String UPDATE_GOLDEN_ARTIFACTS = "UPDATE_GOLDEN_ARTIFACTS";
+  private static final String GOLDEN_FILE_NAME =
+      RepositoryUtilityTest.class.getSimpleName() + "GoldenBomArtifacts.txt";
+  private static final Path GOLDEN_FILE = Paths.get("src", "test", "resources", GOLDEN_FILE_NAME);
+
   @Test
-  public void testReadBom_path() throws MavenRepositoryException, ArtifactDescriptorException {
+  public void testReadBom_path() throws MavenRepositoryException, IOException {
```

Comment:
The purpose of this `testReadBom_path` is to check `RepositoryUtility.readBom(Path)`, not the content of our BOM. Because of the earlier choice of test data `../boms/cloud-oss-bom/pom.xml` for this test, we're now having trouble maintaining the assertion.

If this test just uses a static pom.xml, such as `src/test/resources/libraries-bom-2.2.1-SNAPSHOT.pom` (new file), the problem of maintaining assertion is resolved.

Reply:
Going back to an earlier conversation: what is the purpose of this test?

If it's to test the code that reads a local BOM file to make sure it properly evaluates the indirect BOM references, etc., then a static test file as @suztomo suggests is perfect. It'll never change, and so it's a good data file for a regression test.

If it's to test that any changes to the libraries BOM are intentional and not inadvertent, then a golden file test like this makes sense because it forces code reviewers to approve any change to the effective BOM.

Based on previous discussions with @elharo, I thought that the purpose of the test was the latter. If that's not the case, then the golden file strategy doesn't make sense.

Reply:
It's doing multiple things and perhaps we should split it as @suztomo suggests. The first purpose has always been to test that the `readBom(Path)` method work correctly.